torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -856,9 +1016,62 @@ void SegmentCandidateFinder::findSegments() {
     mergeNodes();
   }
 
+  finalMerge();
+
   finalize();
 }
 
+void SegmentCandidateFinder::finalMerge() {
+  BackwardReachability reachability(segmented_fusion_.get());
+
+  bool merged_nodes = true;
+  while (merged_nodes) {
+    // One merge per iteration
+    for (auto g : groups()) {
```

Comment:
Stick with how you would write. For group in groups. For consumer_edge in consumer_edges. It makes it much easier to read when using words instead of the first letter. If it starts getting really verbose then we can shorten them, but we don't need single letter variables, it's not code golf ;-)

Reply:
Updated, thanks!